apps/ex_nvr/lib/ex_nvr/recordings.ex

```diff
@@ -76,6 +77,48 @@ defmodule ExNVR.Recordings do
     )
   end
 
+  @spec delete_oldest_recordings(Device.t(), integer()) ::
+          {:ok, Ecto.Schema.t()} | {:error, Ecto.Changeset.t()}
+  def delete_oldest_recordings(device, limit) do
+    recordings = Recording.oldest_recordings(device.id, limit) |> Repo.all()
+
+    oldest_recording_query = Recording.oldest_recordings(device.id, 1)
+    delete_recordings_query = from(r in Recording, where: r.id in ^Enum.map(recordings, & &1.id))
+
+    base_run_query = from(r in Run, where: r.device_id == ^device.id)
+    oldest_run_query = from(r in base_run_query, order_by: r.start_date, limit: 1)
+
+    Multi.new()
+    |> Multi.delete_all(:recordings, delete_recordings_query)
+    |> Multi.one(:oldest_recording, oldest_recording_query)
+    |> Multi.delete_all(:runs, fn %{oldest_recording: recording} ->
+      from(r in base_run_query, where: r.end_date < ^recording.start_date)
+    end)
+    |> Multi.one(:oldest_run, oldest_run_query)
+    |> Multi.update(:run, fn %{oldest_run: run, oldest_recording: recording} ->
+      Run.changeset(run, %{start_date: recording.start_date})
+    end)
+    |> Multi.run(:delete_files, fn _repo, _params ->
+      recordings
+      |> Enum.map(&recording_path(device, &1))
+      |> Enum.each(&File.rm!/1)
+
+      {:ok, nil}
+    end)
+    |> Repo.transaction()
+    |> case do
+      {:ok, _params} ->
+        :telemetry.execute([:ex_nvr, :recording, :delete], %{count: length(recordings)}, %{
+          device_id: device.id
+        })
+
+        {:ok, recordings}
```

Comment:
Why are you returning the recordings ??
We don't need them for now, just return `:ok`.

Reply:
I need them for testing, to check that the files are deleted

Reply:
No you don't need them. On the test, you're the one who's creating those records, so you can just check the count on the database and for files, you already have the recordings filenames.

Reply:
A simple test goes as follows:
```elixir
recordings = Enum.map(1..40, ....)
# delete the recordings
# no you can check that the Enum.take(recordings, 30) are deleted.